handlers.js

```diff
@@ -0,0 +1,102 @@
+const isEqual = require('lodash.isequal');
+
+module.exports = self => {
+  return {
+    'apostrophe:modulesRegistered': {
+      async addMissingPieces() {
+        let modified = false;
+
+        const i18nextNamespaces = Object.keys(self.apos.i18n.namespaces)
+          .filter(ns => !self.options.excludeNamespaces.includes(ns));
+
+        const plurals = {
+          plural: 'valuePlural',
+          zero: 'valueZero',
+          two: 'valuePluralTwo',
+          few: 'valuePluralFew',
+          many: 'valuePluralMany'
+        };
+
+        for (const locale of self.apos.i18n.i18next.options.languages) {
+          const req = self.apos.task.getReq({
+            locale,
+            mode: 'draft'
+          });
+          const i18nextResources = i18nextNamespaces.reduce((acc, cur) => {
+            const resources = self.apos.i18n.i18next.getResourceBundle(locale, cur);
+            return {
+              ...acc,
+              ...(resources && { [cur]: resources })
+            };
+          },
+          {});
+
+          const i18nStaticPiecesByNamespace = await self.findPiecesAndGroupByNamespace(`${locale}:draft`);
```

Comment:
This looks correct but the cache document should be used to bypass doing this on every startup.